config_explorer/src/config_explorer/explorer.py

```diff
@@ -0,0 +1,1234 @@
+    """Get the number of replicas and parallelisms.
+
+    Args:
+        report (BenchmarkReport): Benchmark run to evaluate.
+
+    Returns:
+        dict[str, int | None]: Replicas and parallelisms for standalone or
+            prefill/decode configuration. Irrelevant fields will have a value
+            of None.
+    """
+    rp = {
+        'replicas': report.scenario.host.type.count(schema.HostType.REPLICA),
+        'tp': None,
+        'dp': None,
+        'pp': None,
+        'ep': None,
+        'p_replicas': report.scenario.host.type.count(schema.HostType.PREFILL),
+        'p_tp': None,
+        'p_dp': None,
+        'p_pp': None,
+        'p_ep': None,
+        'd_replicas': report.scenario.host.type.count(schema.HostType.DECODE),
+        'd_tp': None,
+        'd_dp': None,
+        'd_pp': None,
+        'd_ep': None,
+    }
+    if rp['replicas'] == 0:
+        rp['replicas'] = None
+    if rp['p_replicas'] == 0:
+        rp['p_replicas'] = None
+    if rp['d_replicas'] == 0:
+        rp['d_replicas'] = None
+
+    if rp['replicas']:
+        # We have an aggregate setup
+        rp['is_pd'] = False
+        rp['tp'] = report.scenario.host.accelerator[0].parallelism.tp
+        rp['dp'] = report.scenario.host.accelerator[0].parallelism.dp
+        rp['pp'] = report.scenario.host.accelerator[0].parallelism.pp
+        rp['ep'] = report.scenario.host.accelerator[0].parallelism.ep
+        return rp
+    # We have a P/D setup
+    rp['is_pd'] = True
+    for ii, accel in enumerate(report.scenario.host.accelerator):
+        if report.scenario.host.type[ii] is schema.HostType.PREFILL and not rp['p_tp']:
+            rp['p_tp'] = accel.parallelism.tp
+            rp['p_dp'] = accel.parallelism.dp
+            rp['p_pp'] = accel.parallelism.pp
+            rp['p_ep'] = accel.parallelism.ep
+        if report.scenario.host.type[ii] is schema.HostType.DECODE and not rp['d_tp']:
+            rp['d_tp'] = accel.parallelism.tp
+            rp['d_dp'] = accel.parallelism.dp
+            rp['d_pp'] = accel.parallelism.pp
+            rp['d_ep'] = accel.parallelism.ep
+        if rp['p_tp'] and rp['d_tp']:
+            break
+    return rp
+
+
+def add_benchmark_report_to_df(
+    runs_df: pd.DataFrame,
+    br_file: str) -> None:
+    """Load a results file and add it to the DataFrame of benchmark runs.
+
+    Args:
+        runs_df (DataFrame): DataFrame to add a row to for the provided run.
+        br_file (str): Benchmark report file to import.
+    """
+    report = convert.import_benchmark_report(br_file)
+
+    # Get plugin parameters
+    prefix_cache_scorer_block_size = None
+    prefix_cache_scorer_lur_capacity_per_server = None
+    prefix_cache_scorer_max_blocks_to_match = None
+    prefix_cache_scorer_tracking = False
+    if report.scenario.platform.metadata:
+        for plugin in report.scenario.platform.metadata['inferenceScheduler']['plugins']:
```

Comment:
Should you first check if `inferenceScheduler` is a key in `.scenario.platform.metadata`? I guess the PD scenario doesn't have `.report.scenario.platform.metadata` section, so we're under the assumption that this script only works for the data on Drive, right?